monai/apps/deepedit/transforms.py

```diff
@@ -95,25 +95,22 @@ def __init__(self, keys: KeysCollection, label_names=None, allow_missing_keys: b
 
         self.label_names = label_names
 
-    def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.ndarray]:
+    def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, NdarrayOrTensor]:
         d: Dict = dict(data)
         for key in self.key_iterator(d):
-            if key == "label":
-                # Dictionary containing new label numbers
-                new_label_names = {}
-                label = np.zeros(d[key].shape)
-                # Making sure the range values and number of labels are the same
-                for idx, (key_label, val_label) in enumerate(self.label_names.items(), start=1):
-                    if key_label != "background":
-                        new_label_names[key_label] = idx
-                        label[d[key] == val_label] = idx
-                    if key_label == "background":
-                        new_label_names["background"] = 0
+            # Dictionary containing new label numbers
+            new_label_names = {}
+            label = torch.zeros_like(d[key])
+            # Making sure the range values and number of labels are the same
+            for idx, (key_label, val_label) in enumerate(self.label_names.items(), start=1):
+                if key_label != "background":
+                    new_label_names[key_label] = idx
+                    label[d[key] == val_label] = idx
+                if key_label == "background":
+                    new_label_names["background"] = 0
 
-                d["label_names"] = new_label_names
-                d[key] = label
-            else:
-                warnings.warn("This transform only applies to the label")
+            d["label_names"] = new_label_names
+            d[key].array = label
```

Comment:
for compatibility this could be
```py
if isinstance(d[key], MetaTensor):
    d[key].array = label
else:
    d[key] = label
```

Reply:
Thanks, @wyli  ... Just updated this bit